frontend/appflowy_flutter/lib/startup/tasks/generate_router.dart

```diff
@@ -0,0 +1,270 @@
+import 'package:appflowy/mobile/presentation/presentation.dart';
+import 'package:appflowy/startup/startup.dart';
+import 'package:appflowy/startup/tasks/app_widget.dart';
+import 'package:appflowy/user/application/auth/auth_service.dart';
+import 'package:appflowy/user/presentation/presentation.dart';
+import 'package:appflowy/workspace/presentation/home/desktop_home_screen.dart';
+import 'package:flowy_infra/time/duration.dart';
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+
+GoRouter generateRouter(Widget child) {
+  return GoRouter(
+    navigatorKey: AppGlobals.rootNavKey,
+    initialLocation: '/',
+    routes: [
+      GoRoute(
```

Comment:
Defining each `GoRoute` as a small function would make the code more readable.

For example,

```dart
  GoRouter(
    routes: [
      _splashRoute,
     _signInRouter
    // ...
   ])

  GoRoute _splashRoute() { return .... }
  GoRoute _signInRouter() { return .... }
```